README.md

```diff
@@ -1 +1,8 @@
-# DesignPatterns
\ No newline at end of file
+
+List of patterns:
+
+1)Factory Pattern
+
+2)Decorator Pattern
+
+3)Iterator Pattern
```

Comment:
HOw does this fit into your assignment? This seems like a general iterator example.